tests/unit/cluster/atomic-slot-migration.tcl

```diff
@@ -1,29 +1,143 @@
-proc migration_status {node_id task_id field} {
-    set status [R $node_id CLUSTER MIGRATION STATUS]
-
-    # Iterate through each migration operation
-    foreach operation $status {
-        set task_id_found ""
-        set field_value ""
-
-        # Parse the key-value pairs in the operation
-        for {set i 0} {$i < [llength $operation]} {incr i 2} {
-            set key [lindex $operation $i]
-            set value [lindex $operation [expr $i + 1]]
-
-            if {$key eq "id"} {
-                set task_id_found $value
-            } elseif {$key eq $field} {
-                set field_value $value
-            }
+set ::slot_prefixes [dict create \
+    0 "{06S}" \
+    1 "{Qi}" \
+    2 "{5L5}" \
+    3 "{4Iu}" \
+    4 "{4gY}" \
+    5 "{460}" \
+    6 "{1Y7}" \
+    7 "{1LV}" \
+    101 "{1j2}" \
+    6000 "{4L7}" \
+    6001 "{4YV}" \
+    6002 "{0bx}" \
+    6003 "{AJ}" \
+    6004 "{of}" \
+]
+
+# Helper functions
+proc get_port {node_id} {
+    if {$::tls} {
+        return [lindex [R $node_id config get tls-port] 1]
+    } else {
+        return [lindex [R $node_id config get port] 1]
+    }
+}
+
+# return the prefix for the given slot
+proc slot_prefix {slot} {
+    return [dict get $::slot_prefixes $slot]
+}
+
+# return a key for the given slot
+proc slot_key {slot {suffix ""}} {
+    return "[slot_prefix $slot]$suffix"
+}
+
+# Populate a slot with keys
+# TODO: Consider merging with populate()
+proc populate_slot {num args} {
+    # Default values
+    set prefix "key:"
+    set size 3
+    set idx 0
+    set prints false
+    set expires 0
+    set slot -1
+
+    # Parse named arguments
+    foreach {key value} $args {
+        switch -- $key {
+            -prefix { set prefix $value }
+            -size { set size $value }
+            -idx { set idx $value }
+            -prints { set prints $value }
+            -expires { set expires $value }
+            -slot { set slot $value }
+            default { error "Unknown option: $key" }
         }
-        # Check if this operation matches the requested task_id
-        if {$task_id_found eq $task_id} {
-            return $field_value
+    }
+
+    # If slot is specified, use slot prefix from table
+    if {$slot >= 0} {
+        if {[dict exists $::slot_prefixes $slot]} {
+            set prefix [dict get $::slot_prefixes $slot]
+        } else {
+            error "Slot $slot not supported in slot_prefixes table, add it manually"
         }
     }
-    # Return empty string if slots_range not found or field not found
-    return ""
+
+    # Try R first (cluster), fallback to r (single instance)
```

Comment:
why do we need this fallback?

Reply:
Not needed actually. I was trying to add slot parameter to `proc populate` but then I gave up for now. Because I'd like to use named parameters as in here e.g. `populate_slot 10000 -idx 1 -slot 6000`   

So, normally this function can be used for non-cluster setups with this fallback line but then we'll need to change ~50-100 other callers of `populate` to make it use named parameters. I didn't do it for now. There is a TODO as a reminder at the top of this proc, so we can revisit this later sometime. I'm deleting this fallback line for now.